database/src/main/postgres/runs/create_partitioning_if_not_exists.sql

```diff
@@ -0,0 +1,108 @@
+    OUT status                  INTEGER,
+    OUT status_text             TEXT,
+    OUT id_partitioning         BIGINT
+) RETURNS record AS
+$$
+    -------------------------------------------------------------------------------
+--
+-- Function: runs.create_partitioning_if_not_exists(3)
+--      Creates a partitioning entry if it does not exist
+--
+-- Parameters:
+--      i_partitioning          - partitioning which existence to check
+--      i_by_user               - user behind the change
+--      i_parent_partitioning   - parent partitioning of the provided partitioning, optional
+--
+-- Returns:
+--      status              - Status code
+--      status_text         - Status text
+--      id_partitioning     - id of the partitioning
+--
+-- Status codes:
+--      11                  - Partitioning created
+--      12                  - Partitioning updated
+--      14                  - Partitioning already present
+--
+-------------------------------------------------------------------------------
+DECLARE
+    _fk_parent_partitioning BIGINT := NULL;
+    _create_partitioning    BOOLEAN;
+    _status                 BIGINT;
+BEGIN
+
+    id_partitioning := runs._get_id_partitioning(i_partitioning, true);
+
+    _create_partitioning := id_partitioning IS NULL;
+
+    IF i_parent_partitioning IS NOT NULL THEN
+        -- TODO check if i_parent_partitioning is parent to i_partitioning #69
+
+        SELECT CPINE.id_partitioning
+        FROM runs.create_partitioning_if_not_exists(i_parent_partitioning, i_by_user, NULL)   CPINE
+        INTO _fk_parent_partitioning;
+    END IF;
+
+
+    IF _create_partitioning THEN
+        INSERT INTO runs.partitionings (partitioning, created_by)
+        VALUES (i_partitioning, i_by_user)
+        RETURNING partitionings.id_partitioning
+        INTO create_partitioning_if_not_exists.id_partitioning;
+
+        PERFORM 1
+        FROM flows._create_flow(id_partitioning, i_by_user);
+
+        status := 11;
+        status_text := 'Partitioning created';
+    ELSE
+        status := 14;
+        status_text := 'Partitioning already present';
+    END IF;
+
+    IF i_parent_partitioning IS NOT NULL THEN
+
+        SELECT ATPF.status
+        FROM flows._add_to_parent_flows(_fk_parent_partitioning, id_partitioning, i_by_user) ATPF
+        INTO _status;
+
+        IF _create_partitioning THEN
+          -- copying measure definitions to establish continuity
+          INSERT INTO runs.measure_definitions(fk_partitioning, measure_name, measured_columns, created_by, created_at)
+          SELECT id_partitioning, CMD.measure_name, CMD.measured_columns, CMD.created_by, CMD.created_at
+          FROM runs.measure_definitions CMD
+          WHERE CMD.fk_partitioning = _fk_parent_partitioning;
+
+          -- additional data are not copied, they are specific for particular partitioning
+        ELSIF (_status = 11) THEN
+          status := 12;
+          status_text := 'Partitioning updated';
```

Comment:
I really don't know whether this is a good sentense - partitioning can't be updated, right? This is a situation when the parent partitioning is added to the flows